cloudweatherreport/model.py

```diff
@@ -521,26 +521,34 @@ def from_bundletester_output(cls, provider, output):
             data = {}
         any_pass = False
         any_fail = False
+        any_infra = False
         for test in data.get('tests', []):
+            lint_or_proof = test['test'] in ('charm-proof', 'make lint')
+            is_exception = 'Traceback' in test['output']
             if test['returncode'] == 0:
                 any_pass = True
+                result_code = 'PASS'
+            elif lint_or_proof and is_exception:
```

Comment:
This could use a comment. I figured it out, but it took me a trip up and down the code. Something along the lines of:

    ```# If the linter or charm proof fail with a traceback, then something went wrong with the infrastructure, rather than with the test.```